### PR TITLE
Add `LAST RESTART` column to `kubectl get pods`

### LINKED ISSUE #54491: kubectl get pods should report last restart time?
When a pod is being restarted, the time it last restarted is very relevant (if it was hours ago, I don't care).  We should consider some information being presented about how recent the restart is.

For example, real world example

```
$ kubectl get pods
NAME           READY     STATUS    RESTARTS   AGE
prometheus-0   5/5       Running   30         21h
```

I don't know when the time of the last restart was.  In this case it was 15 hours ago, so the 30 restarts actually doesn't matter.  The age is 21h, but that's not very useful.  Trying to find a middle ground would be useful.

### PR DESCRIPTION
**Pull request by jjacobelli**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

This PR adds the `LAST RESTART`  column in the output of `kubectl get pods` that represent the time since a container in the pod has restarted.

Example:
```
$ kubectl get pods -A
NAMESPACE                NAME                                    READY   STATUS    RESTARTS   LAST RESTART   AGE
kube-system              coredns-74ff55c5b-6qp7j                 1/1     Running   7          23h            7d3h
kube-system              coredns-74ff55c5b-z79st                 1/1     Running   7          23h            7d3h
kube-system              etcd-jjacobelli-lt                      1/1     Running   6          23h            7d3h
kube-system              kube-apiserver-jjacobelli-lt            1/1     Running   0          <none>         57s
kube-system              kube-controller-manager-jjacobelli-lt   0/1     Running   8          61s            7d3h
kube-system              kube-flannel-ds-c8d66                   1/1     Running   7          8h             7d3h
kube-system              kube-proxy-r9nrx                        1/1     Running   6          23h            7d3h
kube-system              kube-scheduler-jjacobelli-lt            0/1     Running   8          62s            7d3h
```

#### Which issue(s) this PR fixes:

Fixes #54491

#### Special notes for your reviewer:

This is my first contribution to Kubernetes, do not hesitate to tell me what should be done in a different way!

#### Does this PR introduce a user-facing change?

```release-note
kubectl: add "LAST RESTART" column to "kubectl get pods" output
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Ethyling** [comment]
CLA signed

File: pkg/printers/internalversion/printers.go [hunk 1]
```diff
@@ -741,6 +742,7 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 	restarts := 0
 	totalContainers := len(pod.Spec.Containers)
 	readyContainers := 0
+	lastRestartDate := metav1.Date(1, 1, 1, 0, 0, 0, 0, time.UTC)
```
Comment (soltysh) [inline pkg/printers/internalversion/printers.go:745]: `metav1.NewTime(time.Time{})` is simpler

File: pkg/printers/internalversion/printers.go [hunk 2]
```diff
@@ -825,7 +839,12 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 		reason = "Terminating"
 	}
 
-	row.Cells = append(row.Cells, pod.Name, fmt.Sprintf("%d/%d", readyContainers, totalContainers), reason, int64(restarts), translateTimestampSince(pod.CreationTimestamp))
+	lastRestart := "<none>"
+	if !lastRestartDate.IsZero() {
```
Comment (soltysh) [inline pkg/printers/internalversion/printers.go:843]: No need for this condition, `translateTimestampSince` will return `<unknown>` for zero value.

**Review by soltysh — CHANGES_REQUESTED** [review]
Left you some comments.

File: pkg/printers/internalversion/printers.go (hunk 1 above)
Comment (Ethyling) [reply pkg/printers/internalversion/printers.go]: I was looking for something like this, thank you!

File: pkg/printers/internalversion/printers.go (hunk 2 above)
Comment (Ethyling) [reply pkg/printers/internalversion/printers.go]: Good to know, thank you!

**Comment by Ethyling** [comment]
Comments addressed

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by smarterclayton** [comment]
I kind of wish we could combine the column - pods are already very very long.  I.e.:

```
RESTARTS
8 / 24h ago
0
```

**Comment by smarterclayton** [comment]
Historically we have preferred not adding new columns to pods and making existing ones more condensed, because there is a maximum width of most people's shells.

**Comment by soltysh** [comment]
> WDYT @soltysh ?

yeah, I think Clayton's point is a valid one

**Comment by Ethyling** [comment]
I will update the PR then

**Comment by Ethyling** [comment]
Example output using the `RESTARTS` column: 
```
$ kubectl get pods -A
NAMESPACE     NAME                                    READY   STATUS    RESTARTS        AGE
kube-system   coredns-558bd4d5db-lj8kt                1/1     Running   0               17m
kube-system   coredns-558bd4d5db-zq4d2                1/1     Running   0               17m
kube-system   etcd-jjacobelli-ws                      1/1     Running   0               18m
kube-system   kube-apiserver-jjacobelli-ws            1/1     Running   0               64s
kube-system   kube-controller-manager-jjacobelli-ws   1/1     Running   2 / 90s ago     18m
kube-system   kube-flannel-ds-5z8q4                   1/1     Running   0               13m
kube-system   kube-proxy-phv52                        1/1     Running   0               17m
kube-system   kube-scheduler-jjacobelli-ws            1/1     Running   1 / 5m34s ago   18m
```
WDYT @soltysh @smarterclayton ?

**Comment by soltysh** [comment]
> WDYT @soltysh @smarterclayton ?

I personally prefer the 2nd approach so `2 (90s ago)`, let's hear from @smarterclayton

**Comment by soltysh** [comment]
@Ethyling spoke with @smarterclayton on slack, go for the latter and we'll get this merged

**Comment by Ethyling** [comment]
> @Ethyling spoke with @smarterclayton on slack, go for the latter and we'll get this merged

PR updated with the new format

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by smarterclayton** [comment]
/approve

Thanks, looks good (a nice improvement)

**Comment by Ethyling** [comment]
Thank you both!

**Pull request merged**